pkg/controller/cluster/cluster_reconcile.go

```diff
@@ -105,6 +139,9 @@ func (r *ReconcileCluster) Reconcile(request reconcile.Request) (reconcile.Resul
 	}
 
 	helpers.AddTenantLabel(&instance.ObjectMeta, instance.Spec.TenantRef.Name)
+	helpers.AddDeletionProtection(instance)
```

Comment:
What happens if we did the cleanup and removed the finalizer but there's another finalizer still set?
In that case, our finalizer will be added again here and the cleanup will run again on next reconcile. Maybe the `helpers.HandleDeletion` needs to differentiate between: 1. object not deleted, 2. object deleted and our finalizer is set and 3. object deleted but our finalizer is not set.

Reply:
It will now return a struct, which indicates, if the object is deleted and if a finalizer was removed.

So now we can only trigger the actual cleanups once the finalizers are actually removed.